app/models/attachment.rb

```diff
@@ -2,11 +2,22 @@
 
 # entity class for Attachment
 class Attachment < ApplicationRecord
+  include HasPuid
+
+  FORMAT_REGEX = {
+    fasta: /^\S+\.fn?a(sta)?(\.gz)?$/,
+    fastq: /^\S+\.f(ast)?q(\.gz)?$/,
+    text: /^\S+\.(txt|rtf)?(\.gz)?$/,
+    csv: /^\S+\.(csv)?(\.gz)?$/,
+    tsv: /^\S+\.(tsv)?(\.gz)?$/,
+    spreadsheet: /^\S+\.(xls|xlsx)?$/,
+    json: /^\S+\.(json)?(\.gz)?$/,
+    genbank: /^\S+\.(gbk|gbf|genbank)?(\.gz)?$/
+  }.freeze
```

Comment:
Could you also add `{'unknown' => nil}` to this hash. That way format `unknown` is returned through `Attachment::FORMAT_REGEX.keys`